### PR TITLE
Update the routing logic based on recent changes

### PR DESCRIPTION
**Pull request by hross**
### Why:

We have updated the routing logic for runners and want to make it clear in the docs.

### What's being changed:

Routing logic for self hosted runners documentation.

### Check off the following:

- [ ] I have reviewed my changes in staging (look for the latest deployment event in your pull request's timeline, then click **View deployment**).
- [ ] For content changes, I have completed the [self-review checklist](https://github.com/github/docs/blob/main/CONTRIBUTING.md#self-review).

### Writer impact (This section is for GitHub staff members only):

- [ ] This pull request impacts the contribution experience
  - [ ] I have added the 'writer impact' label
  - [ ] I have added a description and/or a video demo of the changes below (e.g. a "before and after video")

### CODE REVIEW

**Comment by ramyaparimi** [comment]
@hross 
Thanks so much for opening a PR! I'll get this triaged for review :zap:

File: content/actions/hosting-your-own-runners/using-self-hosted-runners-in-a-workflow.md [hunk 1]
```diff
@@ -74,5 +74,5 @@ When routing a job to a self-hosted runner, {% data variables.product.prodname_d
 2. The job is then sent to the first matching runner that is online and idle.
```
Comment (TingluoHuang) [inline content/actions/hosting-your-own-runners/using-self-hosted-runners-in-a-workflow.md:74]: {% data variables.product.prodname_dotcom %} first searches for an online and enabled runner at the repository level, then at the organization level{% ifversion ghes or ghae %}, then at the enterprise level{% endif %}.
- If we don't find an online and enabled runner at any level, the job is queued to all levels and wait for any runner from any level to come online and pickup the job.
   - If the job remains queued for more than 24 hours, the job will fail.
- If we find an online and enabled runner (preferred runner) at a certain level, the job is then sent to the preferred runner.
    - 60 seconds after sending the job, if the job is not picked up by the preferred runner, we will try to send the same job to all other levels as well.
    - If the job remains queued for more than 24 hours, the job will fail.

File: content/actions/hosting-your-own-runners/using-self-hosted-runners-in-a-workflow.md [hunk 2]
```diff
@@ -71,8 +71,7 @@ These labels operate cumulatively, so a self-hosted runner’s labels must match
 When routing a job to a self-hosted runner, {% data variables.product.prodname_dotcom %} looks for a runner that matches the job's `runs-on` labels:
 
 1. {% data variables.product.prodname_dotcom %} first searches for a runner at the repository level, then at the organization level{% ifversion ghes or ghae %}, then at the enterprise level{% endif %}. 
+   - If no online runner is found, the job will be queued to all levels and whichever level first has an online and availabile runner will pick up the job.
```
Comment (TingluoHuang) [inline content/actions/hosting-your-own-runners/using-self-hosted-runners-in-a-workflow.md:74]: `If no online and enabled runner is found`

**Review by TingluoHuang — DISMISSED** [review]

File: content/actions/hosting-your-own-runners/using-self-hosted-runners-in-a-workflow.md [hunk 3]
```diff
@@ -70,9 +70,9 @@ These labels operate cumulatively, so a self-hosted runner’s labels must match
...
-   - If all matching runners are offline, the job will queue at the level with the highest number of matching offline runners.
-   - If there are no matching runners at any level, the job will fail.
+- {% data variables.product.prodname_dotcom %} first searches for an online and enabled runner at the repository level, then at the organization level{% ifversion ghes or ghae %}, then at the enterprise level{% endif %}.
+- If {% data variables.product.prodname_dotcom %} doesn't find an online and enabled runner at any level, the job is queued to all levels and waits for any runner from any level to come online and pickup the job.
    - If the job remains queued for more than 24 hours, the job will fail.
+- If {% data variables.product.prodname_dotcom %} finds an online and enabled runner (preferred runner) at a certain level, the job is then sent to the preferred runner.
+    - If the job is not picked up by the preferred runner within 60 seconds after sending the job, {% data variables.product.prodname_dotcom %} will try send the same job to all other levels as well.
```
[the remark this answers is not in the document]
Comment (lucascosti) [reply content/actions/hosting-your-own-runners/using-self-hosted-runners-in-a-workflow.md:77]: What does it mean that `send the same job to all other levels as well.`? Is it the same behavior as the earlier description of `the job is queued to all levels and waits for any runner from any level to come online and pickup the job.`?

Or, is this the same as saying something like: "If the runner doesn't pick up the assigned job within 60 seconds, GitHub starts searching again for an online and enabled runner at all levels."?

File: content/actions/hosting-your-own-runners/using-self-hosted-runners-in-a-workflow.md [hunk 4]
```diff
@@ -70,9 +70,9 @@ These labels operate cumulatively, so a self-hosted runner’s labels must match
...
 
-1. {% data variables.product.prodname_dotcom %} first searches for a runner at the repository level, then at the organization level{% ifversion ghes or ghae %}, then at the enterprise level{% endif %}. 
-2. The job is then sent to the first matching runner that is online and idle.
-   - If all matching online runners are busy, the job will queue at the level with the highest number of matching online runners.
-   - If all matching runners are offline, the job will queue at the level with the highest number of matching offline runners.
-   - If there are no matching runners at any level, the job will fail.
+- {% data variables.product.prodname_dotcom %} first searches for an online and enabled runner at the repository level, then at the organization level{% ifversion ghes or ghae %}, then at the enterprise level{% endif %}.
+- If {% data variables.product.prodname_dotcom %} doesn't find an online and enabled runner at any level, the job is queued to all levels and waits for any runner from any level to come online and pickup the job.
```
Comment (lucascosti) [inline content/actions/hosting-your-own-runners/using-self-hosted-runners-in-a-workflow.md:74]: @TingluoHuang In the previous description, we said that `If there are no matching runners at any level, the job will fail.`. With this new behavior, if there are no runners configured at any level that match the specified labels for the job, will the job be queued and wait 24 hours before failing?

File: content/actions/hosting-your-own-runners/using-self-hosted-runners-in-a-workflow.md [hunk 5]
```diff
@@ -70,9 +70,9 @@ These labels operate cumulatively, so a self-hosted runner’s labels must match
...
    - If the job remains queued for more than 24 hours, the job will fail.
+- If {% data variables.product.prodname_dotcom %} finds an online and enabled runner (preferred runner) at a certain level, the job is then sent to the preferred runner.
+    - If the job is not picked up by the preferred runner within 60 seconds after sending the job, {% data variables.product.prodname_dotcom %} will try send the same job to all other levels as well.
+    - If the job remains queued for more than 24 hours, the job will fail.
```
Comment (lucascosti) [inline content/actions/hosting-your-own-runners/using-self-hosted-runners-in-a-workflow.md:78]: How about we put the match condition first, so we can sum up the queuing and timeout description once at the end. Does this ordering/wording make sense?


```suggestion
- {% data variables.product.prodname_dotcom %} first searches for an online and enabled runner at the repository level, then at the organization level{% ifversion ghes or ghae %}, then at the enterprise level{% endif %}.
- If {% data variables.product.prodname_dotcom %} finds an online and enabled runner at a certain level that matches the job's `runs-on` labels, the job is then assigned and sent to the runner.
    - If the runner doesn't pick up the assigned job within 60 seconds, {% data variables.product.prodname_dotcom %} starts searching again at all levels for an online and enabled runner.
- If {% data variables.product.prodname_dotcom %} doesn't find an online and enabled runner at any level, the job is queued to all levels and waits for any runner from any level to come online and pick up the job.
   - If the job remains queued for more than 24 hours, the job will fail.
```

**Review by lucascosti — COMMENTED** [review]
I asked some questions and made a suggestion to make this a little clearer.

@TingluoHuang When did these change come in to effect? I assume it won't be included in GHES 3.2?

@martin389 We'll probably need to keep the old description for the GHES versions the new one doesn't apply to yet.

File: content/actions/hosting-your-own-runners/using-self-hosted-runners-in-a-workflow.md (hunk 4 above)
Comment (TingluoHuang) [reply content/actions/hosting-your-own-runners/using-self-hosted-runners-in-a-workflow.md:74]: The job will be queued and wait for 24 hours before failing. Within 24 hours, any label matched runner from any level (repo/org/enterprise) that comes online can pick up the job

File: content/actions/hosting-your-own-runners/using-self-hosted-runners-in-a-workflow.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (TingluoHuang) [reply content/actions/hosting-your-own-runners/using-self-hosted-runners-in-a-workflow.md:77]: We don't search for an online and enabled runner after 60 seconds, we queue the job to all levels and wait for a label matched runner from one of the levels that comes online/enable to pick up the job

**Comment by TingluoHuang** [comment]
@lucascosti the change is **NOT** in GHES 3.2

**Comment by lucascosti** [comment]
Ok wording is ready for review:

https://docs-9307--hross-update-assign.herokuapp.com/en/actions/hosting-your-own-runners/using-self-hosted-runners-in-a-workflow#routing-precedence-for-self-hosted-runners

@TingluoHuang / @hross could you please confirm its accuracy?

I've opened a docs-engineering issue internally to look at the check that is failing.

File: content/actions/hosting-your-own-runners/using-self-hosted-runners-in-a-workflow.md [hunk 6]
```diff
@@ -70,9 +70,17 @@ These labels operate cumulatively, so a self-hosted runner’s labels must match
 
 When routing a job to a self-hosted runner, {% data variables.product.prodname_dotcom %} looks for a runner that matches the job's `runs-on` labels:
 
-1. {% data variables.product.prodname_dotcom %} first searches for a runner at the repository level, then at the organization level{% ifversion ghes or ghae %}, then at the enterprise level{% endif %}. 
+{% ifversion fpt or ghes > 3.2 or ghae %}
```
Comment (TingluoHuang) [inline content/actions/hosting-your-own-runners/using-self-hosted-runners-in-a-workflow.md:73]: this behavior is not on for GHAE M1, not sure whether that matters to the doc.

File: content/actions/hosting-your-own-runners/using-self-hosted-runners-in-a-workflow.md [hunk 7]
```diff
@@ -70,9 +70,17 @@ These labels operate cumulatively, so a self-hosted runner’s labels must match
...
+- {% data variables.product.prodname_dotcom %} first searches for an online and idle runner at the repository level, then at the organization level, {% ifversion fpt %} and if the organization is part of an enterprise,{% endif %} then at the enterprise level.
+- If {% data variables.product.prodname_dotcom %} finds an online and idle runner at a certain level that matches the job's `runs-on` labels, the job is then assigned and sent to the runner.
+  - If the runner doesn't pick up the assigned job within 60 seconds, the job is queued at all levels and waits for a matching runner from any level to come online and pick up the job.
+- If {% data variables.product.prodname_dotcom %} doesn't find an online and idle runner at any level, the job is queued to all levels and waits for a matching runner from any level to come online and pick up the job.
+- If the job remains queued for more than 24 hours, the job will fail.
+{% elsif ghes < 3.3 %}
```
Comment (TingluoHuang) [inline content/actions/hosting-your-own-runners/using-self-hosted-runners-in-a-workflow.md:79]: `<= 3.2` ? 😆  I saw the linter error.

**Review by TingluoHuang — DISMISSED** [review]

File: content/actions/hosting-your-own-runners/using-self-hosted-runners-in-a-workflow.md (hunk 6 above)
Comment (lucascosti) [reply content/actions/hosting-your-own-runners/using-self-hosted-runners-in-a-workflow.md:73]: 🤔 Hmm, ok; I'll edit this for `-next`

File: content/actions/hosting-your-own-runners/using-self-hosted-runners-in-a-workflow.md (hunk 7 above)
Comment (lucascosti) [reply content/actions/hosting-your-own-runners/using-self-hosted-runners-in-a-workflow.md:79]: haha, unfortunately, we can't use `<=` or `>=` in our liquid helper 🙁

**Review by martin389 — APPROVED** [review]

**Pull request merged**